### DISCUSSION TITLE
Dynamic gallery, with autoplay enabled, stops after last slide

### DISCUSSION THREAD

**Discussion by CCCullens** [sachinchoolur/lightGallery#1313] [Q&A] [answered]
Hi 
I have a dynamic gallery. Which has 1 to n slides.
I can create a view with autoplay enabled, with the amount of time I want, and turn off user interaction
What I've having problems with is automatically closing the gallery after the last slide is played. 
Should I be leveraging the lgBeforeNextSlide or another event to know I'm at the end. How do I know when I'm at the when that works when there might be 1 or 2 or 100 slides.
Then I would call closeGallery to have the gallery disappear.

Thanks for any suggestions.

**Comment by sachinchoolur** [comment, marked as answer]
Hey @CCCullens,

Yes, you can use, `lgAfterSlide` or similar events. 
You'll get the current slide via event object and you can compare it with the number of slides you have in the gallery and call [`closeGallery`](https://www.lightgalleryjs.com/docs/methods/#closeGallery) function 

Docs - 
https://www.lightgalleryjs.com/docs/events/
https://www.lightgalleryjs.com/docs/methods/

**Reply by CCCullens** [reply]
Thank you!

**Comment by CCCullens** [comment]
Implemented and working well. Was a little more complex for a gallery with only one slide using lgBeforeSlide since next and previous are always both zero. Since my event listener has access to the number of photos, I added code that if 1 slide on lgBeforeSlide, do the equivalent of sleeping for the slideShowInterval then call close Gallery. 

` 
     } else if (galleryPhotos.length === 1) {

                ;(async () => {
                    // Don't close until the slide has been visible for awhile
                    await cc.Sleep(this.ParentCompetition.Preferences.RunThroughPreviewPauseSeconds)
                    lightGalleryView.closeGallery(true)
                })()
            }
`
